src/main/java/com/transferwise/openbanking/client/api/registration/domain/ClientRegistrationRequest.java

```diff
@@ -0,0 +1,54 @@
+package com.transferwise.openbanking.client.api.registration.domain;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.databind.PropertyNamingStrategy;
+import com.fasterxml.jackson.databind.annotation.JsonNaming;
+import com.transferwise.openbanking.client.oauth.domain.GrantType;
+import com.transferwise.openbanking.client.oauth.domain.ResponseType;
+import lombok.Builder;
+import lombok.Data;
+
+import java.util.List;
+
+@Data
+@Builder
+@JsonInclude(JsonInclude.Include.NON_EMPTY)
+@JsonNaming(PropertyNamingStrategy.SnakeCaseStrategy.class)
+public class ClientRegistrationRequest {
```

Comment:
Would it be worth adding javadoc? Is there any other place which explains what field is meant for what? Applies for `ClientRegistrationResponse`

Reply:
I'm hesitant to put javadoc comments on these domain classes as I'm thinking to get rid of them and have them generated dynamically from schema files, the standard publishes the swagger docs so I think it's possible but needs a bit of investigation. Currently our versions of the domain classes often don't have all the possible fields (especially for payments), so switching to schema files approach would resolve that and make it easier to support changes in the standard. 

Having said that I think a class level link to the docs (https://openbankinguk.github.io/dcr-docs-pub/v3.2/dynamic-client-registration.html#obclientregistrationrequest1) would be beneficial and not be any extra hassle so let's do that.